packages/opencensus-core/src/metrics/metric-registry.ts

```diff
@@ -0,0 +1,104 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {checkListElementNotNull, checkNotNull} from '../common/validations';
+import {MeasureUnit} from '../stats/types';
+import {LabelKey} from './export/types';
+
+/**
+ * Creates and manages application's set of metrics.
+ */
+export class MetricRegistry {
+  /**
+   * Builds a new Int64 gauge to be added to the registry. This is more
+   * convenient form when you want to manually increase and decrease values as
+   * per your service requirements.
+   *
+   * @param {string} name The name of the metric.
+   * @param {string} description The description of the metric.
+   * @param {MeasureUnit} unit The unit of the metric.
+   * @param {LabelKey[]} labelKeys The list of the label keys.
+   */
+  addInt64Gauge(
+      name: string, description: string, unit: MeasureUnit,
+      labelKeys: LabelKey[]): void {
+    checkNotNull(name, 'name');
+    checkNotNull(description, 'description');
+    checkNotNull(unit, 'unit');
+    checkListElementNotNull(checkNotNull(labelKeys, 'labelKeys'), 'labelKey');
+    // TODO(mayurkale): Add Int64Gauge.
+  }
+
+  /**
+   * Builds a new double gauge to be added to the registry. This is more
+   * convenient form when you want to manually increase and decrease values as
+   * per your service requirements.
+   *
+   * @param {string} name The name of the metric.
+   * @param {string} description The description of the metric.
+   * @param {MeasureUnit} unit The unit of the metric.
+   * @param {LabelKey[]} labelKeys The list of the label keys.
+   */
+  addDoubleGauge(
+      name: string, description: string, unit: MeasureUnit,
+      labelKeys: LabelKey[]): void {
+    checkNotNull(name, 'name');
+    checkNotNull(description, 'description');
+    checkNotNull(unit, 'unit');
+    checkListElementNotNull(checkNotNull(labelKeys, 'labelKeys'), 'labelKey');
+    // TODO(mayurkale): Add DoubleGauge.
+  }
+
+  /**
+   * Builds a new derived Int64 gauge to be added to the registry. This is more
+   * convenient form when you want to manually increase and decrease values as
+   * per your service requirements.
+   *
+   * @param {string} name The name of the metric.
+   * @param {string} description The description of the metric.
+   * @param {MeasureUnit} unit The unit of the metric.
+   * @param {LabelKey[]} labelKeys The list of the label keys.
+   */
+  addDerivedInt64Gauge(
+      name: string, description: string, unit: MeasureUnit,
+      labelKeys: LabelKey[]): void {
+    checkNotNull(name, 'name');
+    checkNotNull(description, 'description');
+    checkNotNull(unit, 'unit');
+    checkListElementNotNull(checkNotNull(labelKeys, 'labelKeys'), 'labelKey');
```

Comment:
you can simplify it by using `Array.every()`
The same for lines below in this PR

Reply:
```Array.every()``` good option, but it will return boolean values, I have to add another condition to check returned value and throw exception. WDYT?

Reply:
Right - you can use it with `Array.every()` and its builtin function